src/index.ts

```
@@ -197,23 +207,25 @@ export class KeyValueCache<TKeyParams> {
       return false;
     }
 
-    await this.adapter.deleteKeyValue(key);
-    this.entriesCount--;
+    const isDeleted = await this.adapter.deleteKeyValue(key);
+
+    if (!isDeleted) {
+      return false;
+    }
+
+    this.safeDecrementEntriesCount();
 
     if (!value) {
       return true;
+      // We can't decrement the diskSize because the value is falsy.
     }
 
     try {
       const { filePath } = zodValueEntry.parse(JSON.parse(value));
-      const exists = await this.adapter.fileExists(filePath);
-      if (exists) {
-        const fileSize = await this.adapter.fileSize(filePath);
-        this.diskSize -= fileSize;
-        await this.adapter.fileUnlink(filePath);
-      }
+      await this.tryDecrementDiskSize(filePath);
     } catch (err) {
       // Do nothing
+      // We can't decrement the diskSize because the JSON is corrupted.
```

Should this `return true`, as it does on 231 below, or should it actually return false here? (I'm not sure - just asking)

---

Great point! I thought of returning `true` because, although we've failed to parse the JSON or `unlink` the file, we've removed the entry from the KVs. Does this make sense to you?
Maybe we need to surface this error to the client code, so they can decide what to do?